packages/core/src/mask/AbstractMaskSystem.ts

```diff
@@ -10,17 +9,15 @@ import type { Renderer } from '../Renderer';
  * @extends PIXI.System
  * @memberof PIXI
  */
-export class AbstractMaskSystem extends System
+export class AbstractMaskSystem implements ISystem
 {
     protected maskStack: Array<MaskData>;
     protected glConst: number;
     /**
      * @param {PIXI.Renderer} renderer - The renderer this System works for.
      */
-    constructor(renderer: Renderer)
+    constructor(protected renderer: Renderer)
```

Comment:
curious why you have added protected / private in the constructor params?

Reply:
That's a shortcut in TypeScript. These two things are equivalent:

```typescript
constructor(private renderer:Renderer) {}
```

```typescript
private renderer:Renderer;
constructor(renderer:Renderer) {
  this.renderer = renderer;
}
```

Reply:
nice everyday's a school day!